### PR TITLE
Mh/model

### PR DESCRIPTION
**Pull request by holmesm8**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This PR is intended to create two models, the curated cluster (one to many with users) and the curated cluster articles (many-to-many relationship between curated cluster and articles).

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Review by Yetidancer — COMMENTED** [review]
thank you for adding us as contributors!

File: db/migrate/20200512170845_add_user_to_curated_cluster.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 class AddUserToCuratedCluster < ActiveRecord::Migration[5.2]
+  disable_ddl_transaction!
```
Comment (Yetidancer) [inline db/migrate/20200512170845_add_user_to_curated_cluster.rb:2]: have you investigated why rails asked you to put this line in there?

File: db/migrate/20200512170845_add_user_to_curated_cluster.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
   def change
-    add_reference :curated_clusters, :user, foreign_key: true
+    add_reference :curated_clusters, :user, foreign_key: true, index: {algorithm: :concurrently}
```
Comment (Yetidancer) [inline db/migrate/20200512170845_add_user_to_curated_cluster.rb:4]: also curious about the ```index: {algorithm: :concurrently}``` that was suggested

**Comment by ERaith** [comment]
I think we should try to keep our collaboration commits added while we go, or squash the commits to add them in there.

File: db/schema.rb [hunk 3]
```diff
@@ -350,6 +351,23 @@
     t.index ["spent"], name: "index_credits_on_spent"
   end
 
+  create_table "curated_cluster_articles", force: :cascade do |t|
+    t.bigint "article_id"
+    t.datetime "created_at", null: false
+    t.bigint "curated_cluster_id"
+    t.datetime "updated_at", null: false
+    t.index ["article_id"], name: "index_curated_cluster_articles_on_article_id"
+    t.index ["curated_cluster_id"], name: "index_curated_cluster_articles_on_curated_cluster_id"
```
Comment (Yetidancer) [inline db/schema.rb:360]: interesting that these ```t.index```s were added--I have not seen that before

File: db/schema.rb [hunk 4]
```diff
@@ -10,8 +12,7 @@
 #
 # It's strongly recommended that you check this file into your version control system.
 
-ActiveRecord::Schema.define(version: 2020_02_27_214321) do
-
+ActiveRecord::Schema.define(version: 2020_05_12_171425) do
   # These are extensions that must be enabled in order to support this database
```
Comment (ERaith) [inline db/schema.rb:16]: Is the schema always updated when new migrations are created? Or when is the version updated?

**Comment by Yetidancer** [comment]
> I think we should try to keep our collaboration commits added while we go, or squash the commits to add them in there.

I agree--we can start doing this on the client side. It should be easy enough to figure out in vscode

File: db/migrate/20200512170845_add_user_to_curated_cluster.rb (hunk 1 above)
Comment (holmesm8) [reply db/migrate/20200512170845_add_user_to_curated_cluster.rb:2]: Still looking into this. The message came up when trying to commit without line 2 being included. I'm reading through this article that goes over why the line would want to be included:

https://redpanthers.co/disabling-transaction-block-during-migration/

File: db/schema.rb (hunk 4 above)
Comment (Yetidancer) [reply db/schema.rb:16]: Yes, the schema is always updated when new migrations are created. The database, however, is not changed until you db:migrate those changes

File: db/migrate/20200512170845_add_user_to_curated_cluster.rb (hunk 2 above)
Comment (holmesm8) [reply db/migrate/20200512170845_add_user_to_curated_cluster.rb:4]: Same issue as the ddl transaction, this message popped up when attempting to commit the migration file. After looking into this I found a helpful article that explained this process a little further:

https://thoughtbot.com/blog/how-to-create-postgres-indexes-concurrently-in

File: db/schema.rb (hunk 4 above)
Comment (holmesm8) [reply db/schema.rb:16]: Schema's are updated when you run the command "rails db:migrate"

It looks at the migration file, and then updates the schema accordingly.

File: db/schema.rb (hunk 3 above)
Comment (holmesm8) [reply db/schema.rb:360]: My understanding is that these are referencing the foreign keys.

**Pull request merged**